aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt

```diff
@@ -645,7 +646,72 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
         onSuccess: Consumer<MutableList<AuthDevice>>,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        authStateMachine.getCurrentState { authState ->
+            when (authState.authNState) {
+                is AuthenticationState.SignedIn -> {
+                    var token: StateChangeListenerToken? = null
+                    token = credentialStoreStateMachine.listen(
+                        {
+                            when (it) {
+                                is CredentialStoreState.Success -> {
+                                    token?.let(credentialStoreStateMachine::cancel)
+                                    val accessToken = it.storedCredentials?.cognitoUserPoolTokens?.accessToken
+                                    if (!accessToken.isNullOrEmpty()) {
+                                        runBlocking {
```

Comment:
Is it guaranteed that this will run outside of main thread?

Reply:
Changed it to be Async.